code/media/src/test/java/com/adobe/marketing/mobile/media/internal/MediaSessionTests.java

```diff
@@ -521,4 +521,46 @@ public void test_process_NonSessionStartEvent_sessionCreateShouldNotDispatched()
         assertTrue(didSendNetworkRequest());
         assertFalse(mockDispatcherSessionCreated.dispatchSessionCreatedWasCalled);
     }
+
+    @Test
+    public void test_process_retry_connectionIsNull() throws Exception {
+        mediaState.notifyMobileStateChanges(
+                MediaTestConstants.Configuration.SHARED_STATE_NAME, mockData.configSharedState);
+        mediaState.notifyMobileStateChanges(
+                MediaTestConstants.Identity.SHARED_STATE_NAME, mockData.identitySharedState);
+        mediaState.notifyMobileStateChanges(
+                MediaTestConstants.Analytics.SHARED_STATE_NAME, mockData.analyticsSharedState);
+
+        mediaSession.queueHit(mockData.sessionStart);
+        assertEquals(1, mediaSession.getQueueSize());
+
+        // In case of network failure, sessionStart hits are retried 2 more times being dropped.
+        // When network service returns null connection (device offline), hits should should not be
+        // dropped.
```

Comment:
is null connection a (device offline) or (network offline)?

There are two should in this comment.

Reply:
I prefer the wording "network offline".